### PR TITLE
Add types for cli-interact

### PR DESCRIPTION
**Pull request by ffflorian**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

File: types/cli-interact/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export interface NumberOptions {
+    allowNoAnswer?: boolean;
+    requireInteger?: boolean;
+}
+
+export function getChar(promptText: string, allowedCharsAsString: string, flagAllowNoAnswer?: boolean): string;
+export function getChoice(title: string, choices: string[], opts: NumberOptions & {returnNumeric: true}): number;
+export function getChoice(title: string, choices: string[], opts?: ChoiceOptions): string;
+export function getChoiceByChar(title: string, choices: string[], flagAllowNoAnswer?: boolean): string;
```
Comment (amcasey) [inline types/cli-interact/index.d.ts]: Can this return undefined when `flagAllowNoAnswer` is true?

File: types/cli-interact/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export function getInteger(promptText: string): number;
+export function getIPversion(flagAllowNoAnswer?: boolean): string;
+export function getNumber(promptText: string, opts?: boolean | NumberOptions): number;
```
Comment (amcasey) [inline types/cli-interact/index.d.ts]: What does a `Boolean` indicate?  Whether an integer is required?

File: types/cli-interact/index.d.ts (hunk 1 above)
Comment (ffflorian) [reply types/cli-interact/index.d.ts]: No, when `flagAllowNoAnswer` is true, the function returns an empty string.

File: types/cli-interact/index.d.ts (hunk 2 above)
Comment (ffflorian) [reply types/cli-interact/index.d.ts]: Yes. In the source code it says "for historical compatibility".

File: types/cli-interact/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    returnNumeric?: boolean;
+}
+
+export interface NumberOptions {
+    allowNoAnswer?: boolean;
+    requireInteger?: boolean;
+}
+
+export function getChar(promptText: string, allowedCharsAsString: string, flagAllowNoAnswer?: boolean): string;
+export function getChoice(title: string, choices: string[], opts: NumberOptions & {returnNumeric: true}): number;
```
Comment (ffflorian) [inline types/cli-interact/index.d.ts]: I just noticed that this should of course accept `ChoiceOptions` instead of `NumberOptions`. Will update today.

**Pull request merged**